src/backend/routers/student.ts

```diff
@@ -64,6 +64,42 @@ export const student = router({
       return result;
     }),
 
+  /**
+   * Adds a new IEP for the given student.
+   */
+  editIep: authenticatedProcedure
+    .input(
+      z.object({
+        student_id: z.string(),
+        start_date: z.date(),
+        end_date: z.date(),
+      })
+    )
+    .mutation(async (req) => {
+      const { student_id, start_date, end_date } = req.input;
+      const { userId } = req.ctx.auth;
+
+      // Check if the student exists and if the case manager is assigned to the student
+      const existingStudent = req.ctx.db
+        .selectFrom("student")
+        .selectAll()
+        .where("student_id", "=", student_id)
+        .where("assigned_case_manager_id", "=", userId);
+      if (!existingStudent) {
```

Comment:
I could be wrong and I haven't checked this, but if you're using selectAll, then doesn't this query return an array? If so, then I believe that existingStudent will always be truthy as an array with length of 0. I think you want to check if existingStudent.length === 0 rather than the current check, but either way I'd recommend adding tests to make sure that the error is being thrown for a missing or invalid student.

Reply:
`selectAll()` would be selecting all the key:value pairs from all the matches in the db. However, I mentioned above that there might not be any returned data due to not having the `execute()` statement. I do agree with you that checking the length is more accurate.

Reply:
Good call, @KCCPMG and @hieungo89 . I snagged much of this from the addIep function, which has the same issues.

Reply:
FYI, 

how about 

```
if (!existingStudent[0].student_id) {
        throw new Error("Student not found under this Case Manager");
      }
```

since existingStudent =

```
[
  {
    student_id: '9eaa9500-a2a9-4704-b917-7b8c2f28f104',
    first_name: 'student',
    last_name: 'one',
    email: 'one@student.com',
    assigned_case_manager_id: 'a5663b61-691b-49d6-8322-1b3e3f4ba79f',
    grade: 5
  }
]
```

Length checks would be dicey on undefined properties since it may cause a different kind of error.

Reply:
I noted that the major issue was how the seed data was populating the student table, ensuring the assigned_case_manager_id was NOT null.

Reply:
> `selectAll()` would be selecting all the key:value pairs from all the matches in the db. However, I mentioned above that there might not be any returned data due to not having the `execute()` statement. I do agree with you that checking the length is more accurate.

[RE: Kysley](https://kysely.dev/docs/category/select):
```
const persons = await db
  .selectFrom('person')
  .selectAll()
  .execute()
```
translates to:
```
SELECT * FROM "person";
```
You have to have a selectFrom and a select WHAT, even if you're not returning anything. Also, the `execute()` can be any of the execute commands and it will return something, even if you do not need anything returned.

I ended up doing the conditional on `!existingStudent[0]`: please let me know if this works for you.

Reply:
I checked, and the return from `existingStudent` is either array with 0-1 data point. I believe this should be fine. 👍